code/modules/spells/spell_types/rightandwrong.dm

```diff
@@ -146,6 +146,19 @@ GLOBAL_VAR_INIT(summon_magic_triggered, FALSE)
 	var/obj/item/M = new magic_type(get_turf(H))
 	playsound(get_turf(H),'sound/magic/summon_magic.ogg', 50, 1)
 
+	if(M.type == /obj/item/book/granter/spell)

```

Comment:
this needs to use istype() but even doing that the level of spellbook is not updated when it spawns